cloudapi/v1/cloudapi.proto

```diff
@@ -6,49 +6,121 @@ import "google/protobuf/timestamp.proto";
 
 option go_package = "github.com/sourcegraph/cloud-api/go/cloudapi/v1;cloudapiv1";
 
-message ListInstancesRequest {}
+message Instance {
+  string id = 1;
+  optional InstanceDetails instance_details = 2;
+  optional PlatformDetails platform_details = 3;
+  optional InstanceState instance_state = 4;
+}
+
+message InstanceDetails {
+  string name = 1;
+  string type = 2;
+  string environment = 3;
+  string version = 4;
+  optional string full_domain = 5;
```

Comment:
I prefer one of the following:

- `hostname` or `dns_name`, e.g., `feat-1.sgdev.dev` (preferred, cuz it's much easier to work with as consumer. you can transform to other stuff as needed, e.g., add protocol)
- `endpoint`, e.g., `https://feat-1.sgdev.dev`